0022-compartmentalize-grains-via-pop.md

```diff
@@ -0,0 +1,74 @@
+- Feature Name: compartmentalize-grains-via pop
+- Start Date: 2020-06-22
+- SEP Status: Draft
+- SEP PR: https://github.com/saltstack/salt/pull/57681
+- Salt Issue: https://github.com/saltstack/salt/issues/57680
+
+# Summary
+[summary]: #summary
+
+Grains is currently a static component of salt.  We want it to be developed in a more compartmentalize fashion.
+The engineers at SaltStack have created platform-specific projects that functionally replace the grains
+provided by salt with POP.  We propose that this project, codenamed `grainsv2` becomes available in salt while
+maintaining backwards compatibility.
+
+# Motivation
+[motivation]: #motivation
+
+"The original implementation of grains was not designed to scale to the level that salt requires" - thatch45
+Tom wrote POP to solve this problem and we're going to use it. 
+
+By using Plugin Oriented Programming, grains can be separated into individual projects that can be maintained
+by the community safely.
+
+This also solves the issue of `core.py` being an unmaintainable monolith.  It allows platform specific grains
+to be independently maintained.
+
+Working group leaders and other community members focused on specific platforms can manage the release cycle for
+individual idem-platform projects independently from salt.
+
+# Design
+[design]: #detailed-design
+
+Grains are currently created by calling functions in the salt loader.  We would simply need to call
+external POP functions to collect grains.  We need to maintain backwards compatibility with custom user grains.
+
+- All salt grains have already been ported to `grainsv2`.
```

Comment:
Are we sure that all of the changes that have taken place in recent releases are also present in the current pop version?

Reply:
I have been focused on idem-cloud since March, there will need to be a code freeze of future PRs into core.py and changes in the past two months into core.py will need to be merged into the new platform.